### PR TITLE
Refactor codebase

### PR DESCRIPTION
**Pull request by AlexProgrammerDE**

### CODE REVIEW

File: protocol/src/main/java/org/geysermc/mcprotocollib/network/Session.java [hunk 1]
```diff
@@ -33,7 +33,7 @@ public interface Session {
     /**
      * Connects this session to its host and port.
      *
-     * @param wait Whether to wait for the connection to be established before returning.
+     * @param wait         Whether to wait for the connection to be established before returning.
```
Comment (onebeastchris) [inline protocol/src/main/java/org/geysermc/mcprotocollib/network/Session.java]: please revert

File: protocol/src/main/java/org/geysermc/mcprotocollib/network/Session.java [hunk 2]
```diff
@@ -191,7 +191,7 @@ public interface Session {
     /**
      * Sets the compression packet length threshold for this session (-1 = disabled).
      *
-     * @param threshold The new compression threshold.
+     * @param threshold             The new compression threshold.
```
Comment (onebeastchris) [inline protocol/src/main/java/org/geysermc/mcprotocollib/network/Session.java]: please revert

File: protocol/src/main/java/org/geysermc/mcprotocollib/network/packet/PacketProtocol.java [hunk 3]
```diff
@@ -48,7 +48,7 @@ public abstract class PacketProtocol {
     /**
      * Called when a client session is created with this protocol.
      *
-     * @param session The created session.
+     * @param session      The created session.
```
Comment (onebeastchris) [inline protocol/src/main/java/org/geysermc/mcprotocollib/network/packet/PacketProtocol.java]: please revert

File: protocol/src/main/java/org/geysermc/mcprotocollib/protocol/data/game/chunk/DataPalette.java [hunk 4]
```diff
@@ -10,26 +10,13 @@
...
     private BitStorage storage;
-    private final PaletteType paletteType;
-
-    /*
-     * @deprecated globalPaletteBits is no longer in use, use {@link #DataPalette(Palette, BitStorage, PaletteType)} instead.
-     */
-    @Deprecated(forRemoval = true)
-    public DataPalette(@NonNull Palette palette, BitStorage storage, PaletteType paletteType, int globalPaletteBits) {
-        this(palette, storage, paletteType);
-    }
```
Comment (onebeastchris) [inline protocol/src/main/java/org/geysermc/mcprotocollib/protocol/data/game/chunk/DataPalette.java:29]: Why did you remove the deprecation comments here?

File: protocol/src/main/java/org/geysermc/mcprotocollib/protocol/data/game/entity/object/Direction.java [hunk 5]
```diff
@@ -10,16 +10,14 @@ public enum Direction implements ObjectData {
     WEST(2),
     EAST(3);
 
+    public static final Direction[] VALUES = values();
+    private static final Direction[] HORIZONTAL_VALUES = {NORTH, SOUTH, WEST, EAST};
     @Getter
     private final int horizontalIndex;
-
```
Comment (onebeastchris) [inline protocol/src/main/java/org/geysermc/mcprotocollib/protocol/data/game/entity/object/Direction.java:15]: why remove the space between fields and the constructor? we're not doing that elsewhere

**Review by onebeastchris — CHANGES_REQUESTED** [review]

File: protocol/src/main/java/org/geysermc/mcprotocollib/protocol/data/game/chunk/DataPalette.java (hunk 4 above)
Comment (AlexProgrammerDE) [reply protocol/src/main/java/org/geysermc/mcprotocollib/protocol/data/game/chunk/DataPalette.java:29]: It's forRemoval.

Comment (AlexProgrammerDE) [reply protocol/src/main/java/org/geysermc/mcprotocollib/protocol/data/game/chunk/DataPalette.java:29]: dev already requires a major release as package names changed

Comment (onebeastchris) [reply protocol/src/main/java/org/geysermc/mcprotocollib/protocol/data/game/chunk/DataPalette.java:29]: okay, but then also remove the field that's for removal, not just the javadoc that says it's going to be removed

Comment (AlexProgrammerDE) [reply protocol/src/main/java/org/geysermc/mcprotocollib/protocol/data/game/chunk/DataPalette.java:29]: It's still in use, the javadoc is wrong.

Comment (onebeastchris) [reply protocol/src/main/java/org/geysermc/mcprotocollib/protocol/data/game/chunk/DataPalette.java:29]: Where is it currently used except in methods that will also be removed?

Comment (AlexProgrammerDE) [reply protocol/src/main/java/org/geysermc/mcprotocollib/protocol/data/game/chunk/DataPalette.java:29]: sanitizeBitsPerEntry(int)

**Comment by onebeastchris** [comment]
please keep bugfixes/test fixes in separate PRs
